package/kedro_viz/models/flowchart.py

```diff
@@ -93,29 +108,36 @@ class GraphNode(abc.ABC):
     type: str
 
     # the underlying Kedro object for each graph node, if any
-    kedro_obj: Optional[Union[KedroNode, AbstractDataset]] = field(default=None)
+    kedro_obj: Optional[Union[KedroNode, AbstractDataset]]
 
     # the tags associated with this node
-    tags: Set[str] = field(default_factory=set)
+    tags: Set[str] = set()
 
     # the set of registered pipeline IDs this node belongs to
-    pipelines: Set[str] = field(default_factory=set)
+    pipelines: Set[str] = set()
 
     # The original namespace on this node.
     # N.B.: in Kedro, modular pipeline is implemented by declaring namespace on a node.
     # For example, node(func, namespace="uk.de") means this node belongs
     # to the modular pipeline "uk" and "uk.de"
-    namespace: Optional[str] = field(init=False, default=None)
+    namespace: Optional[str]
 
     # The list of modular pipeline this node belongs to.
-    modular_pipelines: Optional[List[str]] = field(init=False)
+    modular_pipelines: Optional[List[str]]
+
+    class Config:
+        """Config class for pydantic GraphNode model to allow
+        arbitrary data types"""
 
-    @staticmethod
-    def _hash(value: str):
+        strict = True
+        arbitrary_types_allowed = True
+
+    @classmethod
+    def _hash(cls, value: str):
         return hashlib.sha1(value.encode("UTF-8")).hexdigest()[:8]
 
-    @staticmethod
-    def _get_namespace(dataset_name: str) -> Optional[str]:
+    @classmethod
```

Comment:
Why is this now a `classmethod` instead of a `staticmethod`? And same for the functions above and below.

Reply:
my bad, there was an error when using staticmethod with pydantic and I thought classmethod is what we use in pydantic. No particular reason and it makes sense to have it as static as it is not relying on any class properties. I will revert this change. Thank you !